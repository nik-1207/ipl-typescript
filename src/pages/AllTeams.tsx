import { Dispatch, SetStateAction, useEffect, useState } from "react";
import axiosCall from "../api/axios";
import LogoMap from "../config/LogoMap";
import Card from "../components/TeamCard";
import ContainerStyle from "../styles/TeamContainerStyle";

interface teamDataType {
  id: string;
  teamName: string;
  venue: string;
  winningYears: [number];
}
function CardContainer() {
  const teamName: string = window.location.pathname.slice(7);
  const style = ContainerStyle();
  const [TeamData, setTeamData]: [unknown, Dispatch<SetStateAction<unknown>>] =
    useState();
  useEffect(() => {
    axiosCall(setTeamData, teamName);
  }, [teamName]);
  const data = TeamData as [teamDataType];
  return (
    <>
      {data && (
        <div className={style.cardcontainer}>
          {Object.keys(LogoMap).map((key: string, index: number) => {
            return (
              <Card
                img={LogoMap[key]}
                key={index}
                id={data[index].id}
                teamName={data[index].teamName}
                venue={data[index].venue}
                winningYears={data[index].winningYears}
              />
            );
          })}
        </div>
      )}
    </>
  );
}

export default CardContainer;
