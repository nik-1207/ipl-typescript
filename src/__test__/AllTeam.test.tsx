import renderer from "react-test-renderer";
import AllTeams from "../pages/AllTeams";

it("navbar snapshot testing", () => {
  const comp = renderer.create(<AllTeams />);
  const snap = comp.toJSON();
  expect(snap).toMatchSnapshot();
});
