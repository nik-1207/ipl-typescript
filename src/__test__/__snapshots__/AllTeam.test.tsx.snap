// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar snapshot testing 1`] = `null`;
